target/cheri-common/cheri_tagmem.c

```diff
@@ -329,9 +299,17 @@ get_iotlb_entry_fast(CPUArchState *env, target_ulong vaddr, int mmu_idx)
      */
 #ifdef CONFIG_DEBUG_TCG
     CPUTLBEntry *entry = tlb_entry(env, mmu_idx, vaddr);
-    g_assert(tlb_hit(tlb_addr_write(entry), vaddr));
+    g_assert(tlb_hit(write ? tlb_addr_write(entry) : entry->addr_read, vaddr));
 #endif
-    return &env_tlb(env)->d[mmu_idx].iotlb[tlb_index(env, mmu_idx, vaddr)];
+    CPUIOTLBEntry *iotlbentry =
+        &env_tlb(env)->d[mmu_idx].iotlb[tlb_index(env, mmu_idx, vaddr)];
+    if (write) {
+        *flags_out = IOTLB_GET_TAGMEM_FLAGS(iotlbentry, write);
```

Comment:
I understand that this is the C preprocessor, so, uh, things are not what they might first seem, but the use of `write`-the-token for token pasting inside the macro and `write`-the-variable on the previous line is a little jarring.  Maybe `bool isWrite`?